src/gwt/src/org/rstudio/studio/client/workbench/views/environment/EnvironmentPresenter.java

```diff
@@ -443,10 +439,24 @@ void onSaveWorkspace()
    {
       view_.bringToFront();
 
-      consoleDispatcher_.saveFileAsThenExecuteCommand("Save Workspace As",
-                                                      ".RData",
-                                                      true,
-                                                      "save.image");
+      server_.isFunctionMasked(
+            "save.image",
+            "base",
+            new ErrorLoggingServerRequestCallback<Boolean>()
+            {
+               public void onResponseReceived(Boolean isMasked)
+               {
+                  String code = isMasked
+                        ? "base::save.image"
+                        : "save.image";
```

Comment:
It seems like the behavior on error here is that we don't save the workspace at all, which seems wrong. Could we use the previous behavior (`save.image`) if we can't determine whether or not the function is masked?